packages/shared/pkg/db/envs.go

```diff
@@ -132,15 +131,15 @@ func (db *DB) GetEnv(ctx context.Context, aliasOrEnvID string) (result *Template
 	if notFound {
 		return nil, nil, fmt.Errorf("template '%s' not found: %w", aliasOrEnvID, err)
 	} else if err != nil {
-		return nil, nil, fmt.Errorf("failed to get env '%s': %w", aliasOrEnvID, err)
+		return nil, nil, fmt.Errorf("failed to get template '%s': %w", aliasOrEnvID, err)
 	}
 
-	build, err = db.Client.EnvBuild.Query().Where(envbuild.EnvID(template.ID), envbuild.StatusEQ(envbuild.StatusUploaded)).Order(models.Desc(envbuild.FieldFinishedAt)).Limit(1).Only(ctx)
+	build, err = db.Client.EnvBuild.Query().Where(envbuild.EnvID(template.ID), envbuild.StatusEQ(envbuild.StatusUploaded)).Order(models.Desc(envbuild.FieldFinishedAt)).First(ctx)
```

Comment:
```
// First returns the first EnvBuild entity from the query.
// Returns a *NotFoundError when no EnvBuild was found.
```

Reply:
I am curious why two queries instead of join?

Reply:
Because entgo doesn't do joins, it does multiple queries underhood and then do the join in memory. This produces better query and also we have better observability and better errors